Refactor to codable swift4

syllerim opened the pull request:
closes #78

syllerim committed:
Remove SwiftyJSON and SwiftyJSON dependencies

syllerim committed:
Refactor code to implement codable protocol

syllerim committed:
refactor code to implement usage of codable on tests

syllerim committed:
remove script to open simulator and clean scheme before installing

alickbass reviewed (approved):
Great job! Looks good!

syllerim committed:
refactor travis script

syllerim committed:
Add test on failure encoding of a model on a network request

syllerim committed:
Fix sintaxis travis script to use platform variables and delete test -f Cartfile

Fix variables definition on travis script

Fix wrong scheme definitions on travis ci script

Remove conditionals to build test on travis script.

change definition xcodebuild action for test or build on travis script

Fix bug action xcodebuild script travis ci

summarize travis script to reuse definition of variables

alickbass reviewed (changes requested):
Can you remove `--verbose` flag from carthage?

alickbass commented on .travis.yml:
```diff
@@ -9,39 +9,22 @@ before_install:
 
 after_success:
-  - bash <(curl -s https://codecov.io/bash)
+  - if [ $TEST == "test" ]; then
```
Do we need this if statement? why not keep it as it used to be?

syllerim committed:
Update Cartfile

syllerim committed:
Add mocking test for Codable model

syllerim committed:
Update readme file

syllerim committed:
Update project and schemes
